crates/bindings-typescript/src/react/useTable.ts

```diff
@@ -0,0 +1,237 @@
+import {
+  useCallback,
+  useEffect,
+  useRef,
+  useState,
+  useSyncExternalStore,
+} from 'react';
+import { useSpacetimeDB } from './useSpacetimeDB';
+import { DbConnectionImpl, TableCache } from '../sdk';
+
+export interface UseQueryCallbacks<RowType> {
+  onInsert?: (row: RowType) => void;
+  onDelete?: (row: RowType) => void;
+  onUpdate?: (oldRow: RowType, newRow: RowType) => void;
+}
+
+type WhereClause = {
+  key: string;
+  operator: '=';
+  value: string | number | boolean;
+};
+
+export function eq<ColumnType extends string>(
+  key: ColumnType,
+  value: string | number | boolean
+): WhereClause {
+  return { key, operator: '=', value };
+}
+
+export function where(condition: WhereClause) {
+  return `${condition.key} ${condition.operator} ${JSON.stringify(condition.value)}`;
+}
+
+function matchesWhereClause(
+  row: Record<string, any>,
+  whereClause: WhereClause
+): boolean {
+  const { key, operator, value } = whereClause;
+  if (!(key in row)) {
+    return false;
+  }
+  switch (operator) {
+    case '=':
+      return row[key] === value;
+    default:
+      return false;
+  }
+}
+
+type Snapshot<RowType> = {
+  readonly rows: readonly RowType[];
+  readonly state: 'loading' | 'ready';
+};
+
+export function useTable<
+  DbConnection extends DbConnectionImpl,
+  RowType extends Record<string, any>,
+  TableName extends keyof DbConnection['db'] &
+    string = keyof DbConnection['db'] & string,
+>(
+  tableName: TableName,
+  whereClauseOrCallbacks?: WhereClause | UseQueryCallbacks<RowType>,
+  callbacks?: UseQueryCallbacks<RowType>
+): Snapshot<RowType> {
+  let whereClause: WhereClause | undefined;
+  if (whereClauseOrCallbacks && 'key' in whereClauseOrCallbacks) {
+    whereClause = whereClauseOrCallbacks;
+  } else {
+    callbacks = whereClauseOrCallbacks as
+      | UseQueryCallbacks<RowType>
+      | undefined;
+  }
+  const [subscribeApplied, setSubscribeApplied] = useState(false);
+  const [isActive, setIsActive] = useState(false);
+  let spacetime: DbConnection | undefined;
+  try {
+    spacetime = useSpacetimeDB<DbConnection>();
+  } catch {
+    throw new Error(
+      'Could not find SpacetimeDB client! Did you forget to add a' +
+        '`SpacetimeDBProvider`? `useTable` must be used in the React component tree' +
```

Comment:
```suggestion
      'Could not find SpacetimeDB client! Did you forget to add a ' +
        '`SpacetimeDBProvider`? `useTable` must be used in the React component tree' +
```
I think it's lacking a space between the two concatenated string "a`SpacetimeDBProvider`"

Reply:
Oh very good catch!